### PR TITLE
Always stop services during state clean (config|cache|uninstall)

### PR DESCRIPTION
**Pull request by mdrohmann**
https://www.pivotaltracker.com/story/show/178564019

### CODE REVIEW

File: internal/runners/clean/stop.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	// Todo: https://www.pivotaltracker.com/story/show/177585085
+	// Yes this is awkward right now
+	if err := installation.StopTrayApp(cfg); err != nil {
+		if ignoreErrors {
+			return errs.AddTips(
+				locale.WrapError(err, "clean_stop_tray_failure", "Cleanup interrupted, because a running {{.V0}} process could not be stopped.", trayInfo.Name()),
+				cleanForceTip)
+		} else {
+			out.Print(locale.Tl("clean_stop_tray_warning", "[ERROR]Failed to stop running {{.V0}} process.[/RESET] Continuing anyways, because --force flag was provided.", trayInfo.Name()))
+		}
```
Comment (daved) [inline internal/runners/clean/stop.go:29]: I suggest making this !ignoreErrors and dropping the else. And the same below.

File: internal/runners/clean/run_win.go [hunk 2]
```diff
@@ -66,28 +66,7 @@ func removeConfig(configPath string, out output.Outputer) error {
...
-		exitCode, _, err := exeutils.Execute(svcInfo.Exec(), []string{"stop"}, nil)
-		if err != nil {
-			return errs.Wrap(err, "Stopping %s returned error", svcInfo.Name())
-		}
-		if exitCode != 0 {
-			return errs.New("Stopping %s exited with code %d", svcInfo.Name(), exitCode)
-		}
-	}
-
+func removeInstall(cfg configurable, logFile, configPath string) error {
```
Comment (daved) [inline internal/runners/clean/run_win.go:66]: configurable might be able to be dropped from the params.

**Review by daved — CHANGES_REQUESTED** [review]

File: internal/runners/clean/stop.go (hunk 1 above)
Comment (mdrohmann) [reply internal/runners/clean/stop.go:29]: :facepalm:

**Review by daved — APPROVED** [review]

File: internal/runners/clean/stop.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	// Stop state-svc before accessing its files
+	if fileutils.FileExists(svcInfo.Exec()) {
+		_, _, err := exeutils.Execute(svcInfo.Exec(), []string{"stop"}, nil)
+		if err != nil {
+			if !ignoreErrors {
+				return errs.AddTips(
+					locale.WrapError(err, "clean_stop_svc_failure", "Cleanup interrupted, because a running {{.V0}} process could not be stopped.", svcInfo.Name()),
+					cleanForceTip)
+			}
+			out.Print(locale.Tl("clean_stop_svc_warning", "[ERROR]Failed to stop running {{.V0}} process.[/RESET] Continuing anyways, because --force flag was provided.", svcInfo.Name()))
```
Comment (Naatan) [inline internal/runners/clean/stop.go:40]: I think this can be `out.Error()` and you can drop the `[ERROR]`

**Review by Naatan — CHANGES_REQUESTED** [review]

**Review by Naatan — APPROVED** [review]

**Pull request merged**